### ISSUE TITLE
KeyError "resources/hook-tmpl" when running "pre-commit-3.6.0.pyz" in Windows

### ISSUE THREAD

**Issue by Panther-1ne** [pre-commit/pre-commit#3116]
### search you tried in the issue tracker

KeyError

### describe your issue

Hi,
I am a pre-commit newbie. First of all, thank you for this great tool! 👍

I would like to report what looks like a bug specifically in the 0-dependency version `pre-commit-3.6.0.pyz`. I have a cross-platform build project. First I tried out pre-commit on Linux and everything worked as expected. Than I tried it on Windows (Git bash and cmd), and it kept complaining: 
```
An unexpected error has occurred: KeyError: "There is no item named 'pre_commit\\\\resources/hook-tmpl' in the archive"
```
I thought there was something wrong with my python setup (python3.12) but then I tried an older version of the pyz (pre-commit-3.5.0.pyz) and that worked just fine. Is there a file missing in 3.6.0?

Thanks,
Panther

### pre-commit --version

pre-commit-3.6.0

### .pre-commit-config.yaml

```yaml
repos:
  - repo: https://github.com/pre-commit/pre-commit-hooks
    rev: v3.2.0
    hooks:
      - id: trailing-whitespace
        args: [ --markdown-linebreak-ext=md ]
        description: trailing whitespace allowed in .md files only
      - id: end-of-file-fixer
      - id: mixed-line-ending
```

### version information

```
pre-commit version: 3.6.0
git --version: git version 2.39.2.windows.1
sys.version:
    3.9.13 (tags/v3.9.13:6de2ca5, May 17 2022, 16:36:42) [MSC v.1929 64 bit (AMD64)]
sys.executable: C:\Users\panther\.cache\pre-commit-zipapp\bTYEHS-1SlfKMYxvfi4yWi3oQe7Id2hN6Tj0kyvFIWE\python.exe
os.name: nt
sys.platform: win32
```

### error information

```
An unexpected error has occurred: KeyError: "There is no item named 'pre_commit\\\\resources/hook-tmpl' in the archive"
```

```
Traceback (most recent call last):
  File "C:\Users\panther\.cache\pre-commit-zipapp\bTYEHS-1SlfKMYxvfi4yWi3oQe7Id2hN6Tj0kyvFIWE\wheels\pre_commit-3.6.0-py2.py3-none-any.whl\pre_commit\error_handler.py", line 73, in error_handler
    yield
  File "C:\Users\panther\.cache\pre-commit-zipapp\bTYEHS-1SlfKMYxvfi4yWi3oQe7Id2hN6Tj0kyvFIWE\wheels\pre_commit-3.6.0-py2.py3-none-any.whl\pre_commit\main.py", line 396, in main
    return install(
  File "C:\Users\panther\.cache\pre-commit-zipapp\bTYEHS-1SlfKMYxvfi4yWi3oQe7Id2hN6Tj0kyvFIWE\wheels\pre_commit-3.6.0-py2.py3-none-any.whl\pre_commit\commands\install_uninstall.py", line 131, in install
    _install_hook_script(
  File "C:\Users\panther\.cache\pre-commit-zipapp\bTYEHS-1SlfKMYxvfi4yWi3oQe7Id2hN6Tj0kyvFIWE\wheels\pre_commit-3.6.0-py2.py3-none-any.whl\pre_commit\commands\install_uninstall.py", line 93, in _install_hook_script
    contents = resource_text('hook-tmpl')
  File "C:\Users\panther\.cache\pre-commit-zipapp\bTYEHS-1SlfKMYxvfi4yWi3oQe7Id2hN6Tj0kyvFIWE\wheels\pre_commit-3.6.0-py2.py3-none-any.whl\pre_commit\util.py", line 40, in resource_text
    return files.joinpath(filename).read_text()
  File "C:\Program Files (x86)\Microsoft Visual Studio\Shared\Python39_64\lib\zipfile.py", line 2327, in read_text
    with self.open('r', *args, **kwargs) as strm:
  File "C:\Program Files (x86)\Microsoft Visual Studio\Shared\Python39_64\lib\zipfile.py", line 2315, in open
    stream = self.root.open(self.at, zip_mode, pwd=pwd)
  File "C:\Program Files (x86)\Microsoft Visual Studio\Shared\Python39_64\lib\zipfile.py", line 1511, in open
    zinfo = self.getinfo(name)
  File "C:\Program Files (x86)\Microsoft Visual Studio\Shared\Python39_64\lib\zipfile.py", line 1438, in getinfo
    raise KeyError(
KeyError: "There is no item named 'pre_commit\\\\resources/hook-tmpl' in the archive"
```

**Comment by asottile** [comment]
at a glance this looks like a bug in importlib-resources which would point to cpython being broken

**Comment by Panther-1ne** [comment]
Yes, but **I am not sure if I did it right** (I am C++ guy).. I built CPython on Windows (Visual Studio 2022). And tried the following:
```
pythonw_d.exe pre-commit-3.6.0.pyz install -f
```
Once again version 3.5.0 generated an expected output. But 3.6.0 generated a zero-byte script `pre-commit`. The log file looks similar to the last one:

### version information

```
pre-commit version: 3.6.0
git --version: git version 2.39.2.windows.1
sys.version:
    3.9.13 (tags/v3.9.13:6de2ca5, May 17 2022, 16:36:42) [MSC v.1929 64 bit (AMD64)]
sys.executable: C:\Users\panther\.cache\pre-commit-zipapp\bTYEHS-1SlfKMYxvfi4yWi3oQe7Id2hN6Tj0kyvFIWE\python.exe
os.name: nt
sys.platform: win32
```

### error information

```
An unexpected error has occurred: KeyError: "There is no item named 'pre_commit\\\\resources/hook-tmpl' in the archive"
```

```
Traceback (most recent call last):
  File "C:\Users\panther\.cache\pre-commit-zipapp\bTYEHS-1SlfKMYxvfi4yWi3oQe7Id2hN6Tj0kyvFIWE\wheels\pre_commit-3.6.0-py2.py3-none-any.whl\pre_commit\error_handler.py", line 73, in error_handler
    yield
  File "C:\Users\panther\.cache\pre-commit-zipapp\bTYEHS-1SlfKMYxvfi4yWi3oQe7Id2hN6Tj0kyvFIWE\wheels\pre_commit-3.6.0-py2.py3-none-any.whl\pre_commit\main.py", line 396, in main
    return install(
  File "C:\Users\panther\.cache\pre-commit-zipapp\bTYEHS-1SlfKMYxvfi4yWi3oQe7Id2hN6Tj0kyvFIWE\wheels\pre_commit-3.6.0-py2.py3-none-any.whl\pre_commit\commands\install_uninstall.py", line 131, in install
    _install_hook_script(
  File "C:\Users\panther\.cache\pre-commit-zipapp\bTYEHS-1SlfKMYxvfi4yWi3oQe7Id2hN6Tj0kyvFIWE\wheels\pre_commit-3.6.0-py2.py3-none-any.whl\pre_commit\commands\install_uninstall.py", line 93, in _install_hook_script
    contents = resource_text('hook-tmpl')
  File "C:\Users\panther\.cache\pre-commit-zipapp\bTYEHS-1SlfKMYxvfi4yWi3oQe7Id2hN6Tj0kyvFIWE\wheels\pre_commit-3.6.0-py2.py3-none-any.whl\pre_commit\util.py", line 40, in resource_text
    return files.joinpath(filename).read_text()
  File "C:\Program Files (x86)\Microsoft Visual Studio\Shared\Python39_64\lib\zipfile.py", line 2327, in read_text
    with self.open('r', *args, **kwargs) as strm:
  File "C:\Program Files (x86)\Microsoft Visual Studio\Shared\Python39_64\lib\zipfile.py", line 2315, in open
    stream = self.root.open(self.at, zip_mode, pwd=pwd)
  File "C:\Program Files (x86)\Microsoft Visual Studio\Shared\Python39_64\lib\zipfile.py", line 1511, in open
    zinfo = self.getinfo(name)
  File "C:\Program Files (x86)\Microsoft Visual Studio\Shared\Python39_64\lib\zipfile.py", line 1438, in getinfo
    raise KeyError(
KeyError: "There is no item named 'pre_commit\\\\resources/hook-tmpl' in the archive"
```

**Comment by Panther-1ne** [comment]
PS: sorry for picking Friday :) to report an issue

**Comment by asottile** [comment]
you probably don't want pythonw but otherwise yes that is the correct usage

**Comment by asottile** [comment]
ok so the output above shows python 3.9.13 -- I've verified it's also broken in that version, but fixed in python 3.10 -- so you'll need to use a newer version of python to run the `.pyz` (the bug has already been fixed by cpython)
